packages/components/src/higher-order/with-fallback-styles/index.js

```diff
@@ -38,9 +40,21 @@ export default ( mapNodeToProps ) => createHigherOrderComponent(
 				this.grabFallbackStyles();
 			}
 
+			shouldRecomputeValues() {
+				if ( ! computeWhenChange ) {
+					return ! this.state.grabStylesCompleted;
+				}
+				const newRecomputeValues = computeWhenChange( this.props );
+				if ( isShallowEqual( this.lastRecomputeValues, newRecomputeValues ) ) {
+					return false;
+				}
+				this.lastRecomputeValues = newRecomputeValues;
```

Comment:
Is there any alternative way to implement it which would avoid using instance variable? In general, it's been always very hard to make it work in a predictable way in other parts of Gutenberg.

Reply:
Hi @gziolo this PR does not use instance variables now. We take advantage of the function scope which is more predictable, only calls to shouldRecomputeValues use and update lastRecomputeValues. The other possible alternative is using state but it would cause unnecessary rerenders.